specific/cmo_project/models/project.py

```diff
@@ -293,15 +315,116 @@ def _get_state_before_inactive(self):
+                project.project_parent_id.id)
             project.parent_id = parent_project.analytic_account_id.id
 
     @api.multi
     @api.constrains('brief_date', 'date')
     def _check_brief_dates(self):
         self.ensure_one()
         if self.brief_date > self.date:
-            return ValidationError("project brief-date must be lower than project end-date.")
+            return ValidationError("project brief-date must be lower than \
+                project end-date.")
+
+    @api.multi
+    def quotation_relate_project_tree_view(self):
+        self.ensure_one()
+        domain = [
+            '&',
+            ('project_related_id', 'like', self.id),
+            ('order_type', '=', 'quotation'),
+        ]
+        return {
+            'name': 'Quotations',
+            'res_model': 'sale.order',
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'tree,form',
+            'domain': domain,
+            'context': "{'active': True, 'order_type': 'quotation'}",
+        }
+
+    @api.multi
+    def purchase_relate_project_tree_view(self):
+        self.ensure_one()
+        domain = [
+            ('project_id', 'like', self.id),
+        ]
+        return {
+            'name': 'Purchase Order',
+            'res_model': 'purchase.order',
+            'type': 'ir.actions.act_window',
+            'views': [[False, "tree"], [False, "form"]],
+            'domain': domain,
+            'context': "{'active': True}"
+        }
+
+    @api.multi
+    @api.depends(
+        'actual_price',
+        'estimate_cost',
+        'quote_related_ids',
+        'quote_related_ids.order_line',
+    )
+    def _compute_price_and_cost(self):
+        for project in self:
+            actual_price = 0
+            estimate_cost = 0
+            quotes = self.quote_related_ids.filtered(
+                lambda r: r.state in ('draft', 'done')
+            )
+            for quote in quotes:
+                actual_price += quote.amount_untaxed
+                estimate_cost += sum(quote.order_line.filtered(
+                    lambda r: r.purchase_price > 0).mapped('purchase_price'))
+            project.actual_price = actual_price
+            project.estimate_cost = estimate_cost
+
+    @api.multi
+    @api.depends(
+        'actual_po',
+        'purchase_related_ids',
+        'purchase_related_ids.state',
+        'purchase_related_ids.amount_untaxed',
+    )
+    def _compute_actual_po(self):
+        for project in self:
+            purchase_orders = project.purchase_related_ids.filtered(
+                lambda r: r.state in ('approved', 'done')
+            )
+            project.actual_po = sum(purchase_orders.mapped('amount_untaxed'))
+
+    @api.multi
+    def name_get(self):
+        res = []
+        for project in self:
+            name = project.name or '/'
+            if name and project.project_number:
+                name = name + ' ('+project.project_number+')'
```

Comment:
Always use string injection over "+" to concatenate string, i..e,
`'%s (%s)' % (name, project.project_number)`